### PR TITLE
Add ensembl-utils and update documentation generation

### PR DESCRIPTION
**Pull request by JAlvarezJarreta**
I have also bumped minimum version to 3.10.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,30 +45,27 @@ If you need to install "editable" python package use '-e' option
...
 
 
-### Additional steps to use automated generation of the documentation (part of it)
-Install python part with the `[doc]` tag.
-Change into repo dir
-Run doc build script.
+### Additional steps to use automated generation of the documentation
+- Install python part with the `[docs]` tag
+- Change into repo dir
+- Run dos build script
```
Comment (ens-LCampbell) [inline README.md:59]: typo ?

File: pyproject.toml [hunk 2]
```diff
@@ -35,22 +35,27 @@ keywords = [
...
     "Intended Audience :: Science/Research",
     "License :: OSI Approved :: Apache Software License",
-    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
```
Comment (ens-LCampbell) [inline pyproject.toml:48]: Should we state before testing 3.10/3.11 ?

File: .github/workflows/mkdocs_docs_generation.yml [hunk 3]
```diff
@@ -25,50 +25,34 @@ on:
...
-    
-    - name: Get pip cache dir 
-      env:
-        PIPCACHE: ${{ steps.pip-cache-dir-path.outputs.PIPCACHE }}
-      run: echo "The pip cache dir located is $PIPCACHE"
+        python -m pip install --upgrade pip
+        pip install -e .[docs]
 
-    - name: Install Dependencies
+    - name: Run mkdocs
```
Comment (ens-LCampbell) [inline .github/workflows/mkdocs_docs_generation.yml:55]: Its so bare bones now XD Much more streamlined !

**Review by ens-LCampbell — APPROVED** [review]
Looks ok to me Jorge :)

File: pyproject.toml (hunk 2 above)
Comment (JAlvarezJarreta) [reply pyproject.toml:48]: The way we have set the CI pipeline conditions does not allow for manual runs on branches that are not `main` of `hackathon/*`, so it cannot be done before that. However, I did test 3.10 some time ago and it was ok. I doubt 3.11 will be different.

**Comment by JAlvarezJarreta** [comment]
Python 3.10 has been successful as well:
* https://gitlab.ebi.ac.uk/vectorbase/ensembl-genomio/-/pipelines/527696

However, I'm not sure why Python 3.11 is failing on a single parametrization of `assembly/test_download.py` (have tried to fix it unsuccessfully):
* https://gitlab.ebi.ac.uk/vectorbase/ensembl-genomio/-/pipelines/527675

**Comment by JAlvarezJarreta** [comment]
Python 3.11 is happy as well now:
* https://gitlab.ebi.ac.uk/vectorbase/ensembl-genomio/-/pipelines/527711

**Review by Dishalodha — APPROVED** [review]

**Review by ens-LCampbell — APPROVED** [review]
Recent updates to PR look A-ok!

**Pull request merged**